src/solver/image_transform/adjust_hue.cpp

```diff
@@ -0,0 +1,140 @@
+#include <vector>
+
+namespace miopen {
+namespace solver {
+namespace image_transform {
+namespace adjust_hue {
+
+bool ImageAdjustHue::IsApplicable(
+    const ExecutionContext& /* context */,
+    const miopen::image_transform::adjust_hue::ProblemDescription& problem) const
+{
+    if(!problem.IsSameType())
+        return false;
+    if(!problem.IsImprovementOverROCm())
+        return false;
+    return true;
+}
+
+ConvSolution ImageAdjustHue::GetSolution(
+    const ExecutionContext& /* context */,
+    const miopen::image_transform::adjust_hue::ProblemDescription& problem) const
+{
+    auto result = ConvSolution(miopenStatusSuccess);
+
+    auto dtype   = problem.GetInputTensorDesc().GetType();
+    auto io_type = miopen::GetDataType(dtype);
+    auto numel   = problem.GetInputTensorDesc().GetElementSize();
+
+    auto input_size = numel / 3; // RGB image
+
+    size_t xlocalsize = 256;
+    size_t xgridsize  = AlignUp(input_size, xlocalsize);
+
+    size_t ylocalsize = 1;
+    size_t ygridsize  = 1;
+
+    size_t zlocalsize = 1;
+    size_t zgridsize  = 1;
+
+    auto kernel = KernelInfo{};
+
+    kernel.kernel_file = "MIOpenImageAdjustHue.cpp";
+    if(problem.GetInputTensorDesc().IsContiguous() && problem.GetOutputTensorDesc().IsContiguous())
+        kernel.kernel_name = "ImageAdjustHueContiguous";
+    else
+        kernel.kernel_name = "ImageAdjustHue";
+
+    const auto build_params =
+        KernelBuildParameters{{"MIOPEN_USE_FP16", static_cast<int32_t>(dtype == miopenHalf)},
+                              {"MIOPEN_USE_FP32", static_cast<int32_t>(dtype == miopenFloat)},
+                              {"MIOPEN_USE_FP64", static_cast<int32_t>(dtype == miopenDouble)},
+                              {"MIOPEN_USE_BFP16", static_cast<int32_t>(dtype == miopenBFloat16)}};
+
+    kernel.comp_options = build_params.GenerateFor(kbp::HIP{});
+
+    kernel.l_wk.push_back(xlocalsize);
+    kernel.l_wk.push_back(ylocalsize);
+    kernel.l_wk.push_back(zlocalsize);
+
+    kernel.g_wk.push_back(xgridsize);
+    kernel.g_wk.push_back(ygridsize);
+    kernel.g_wk.push_back(zgridsize);
+
+    result.construction_params.push_back(kernel);
+
+    result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+            decltype(auto) kernel = handle_.Run(kernels.front());
+            decltype(auto) params =
+                raw_params.CastTo<miopen::image_transform::adjust_hue::InvokeParams>();
+
+            auto xdesc = miopen::deref(params.inputTensorDesc);
+            auto ydesc = miopen::deref(params.outputTensorDesc);
+
+            auto x_tv = get_inner_expanded_4d_tv(xdesc);
+            auto y_tv = get_inner_expanded_4d_tv(ydesc);
+
+            size_t N        = xdesc.GetElementSize() / 3;
+            size_t c_stride = xdesc.GetLengths()[2] * xdesc.GetLengths()[3];
+
+            if(kernel.name == "ImageAdjustHueContiguous")
```

Comment:
I mean either way it is going to make two `invoker_factory` object anyway, so I doubt it is going to have much impact. If necessary, I can perform the necessary changes, but since these kernels have really similar launch parameters, I think this approach would be cleaner and avoids code duplication.

Also, this is exactly how it is implemented in MoDNN